### PR TITLE
Added Ignore functionality to Script Table

### PR DESCRIPTION
**Pull request by eireglas**
If the fixture returns 'ignore' as a string value then that test and all subsequent rows in the scriptTable are set to ignore and has 'Test not run' displayed. See below example:
image

### CODE REVIEW

**Comment by jediwhale** [comment]
Is this a breaking change, if an existing fixture returns "ignore" as a valid value?

**Comment by eireglas** [comment]
I agree with your suggestion. Returning an exception would be better, I will refactor 👍

**Comment by fhoeben** [comment]
This just works at the script table level. It would be good if it were similar to the existing 'StopTest' and 'StopSuite' exception handling (which abort a page or the entire suite respectively). And please also add some documentation for the new feature.

**Comment by jediwhale** [comment]
I think adding it to Script table is a good start.

**Comment by fhoeben** [comment]
I suppose it is good to add it to 

 * FitNesse.FullReferenceGuide.UserGuide.WritingAcceptanceTests.SliM.SlimProtocol, the 'Aborting a Test' section
 * FitNesse.UserGuide.WritingAcceptanceTests.SliM.ExceptionHandling, the 'Stopping the execution of the test.' section

**Comment by eireglas** [comment]
@fhoeben I've updated the documentation and fixed a unit test.

**Comment by eireglas** [comment]
@fhoeben @jediwhale could either of you review and approve, if happy? Thanks.

File: FitNesseRoot/FitNesse/UserGuide/WritingAcceptanceTests/SliM/ExceptionHandling/content.txt [hunk 1]
```diff
@@ -8,6 +8,10 @@ If your exception indicates an error that cannot be recovered from, then you may
 
 By throwing an exception with "!-StopSuite-!" in the class name you can stop both current test and the suite execution. Tear down procedure for the running test and the suite will still be executed, but all the consequent tests in the suite will not run.
 
+!2 Ignoring Tests during execution
+If you want to ignore or skip certain parts of your test during execution you can do this by throwing an ignore exception. By throwing an exception with "!-IgnoreScriptTest-!" in the class name you can ignore and skip the instructions in the rest of that script table.
+If you want to ignore or skip all subsequent parts of the current batch during execution you can do this by throwing an ignore all exception. By throwing an exception with "!-IgnoreAllTests-!" in the class name you can ignore and skip all instructions in the current batch.
```
Comment (fhoeben) [inline FitNesseRoot/FitNesse/UserGuide/WritingAcceptanceTests/SliM/ExceptionHandling/content.txt:13]: What do you mean with 'current batch', the current test page, or suite, or ...?

File: FitNesseRoot/FitNesse/UserGuide/WritingAcceptanceTests/SliM/SlimProtocol/content.txt [hunk 2]
```diff
@@ -146,6 +146,12 @@ If a fixture throws an exception with a ''class'' name that contains "!style_cod
 
 To abort the test suite (stop execution directly), throw an exception with a class name that contains "!style_code(!-StopSuite-!)". The syntax is identical to !-StopTest-!.
 
+!4 Ignoring Test
+If a fixture throws an exception with a ''class'' name that contains "!style_code(!-IgnoreScriptTest-!)", then Slim should ignore and skip the instructions in the rest of that script table.   The response for this type of exception should be "!style_code(__EXCEPTION__:IGNORE_SCRIPT_TEST:)" which may have an optional suffix of: "!style_code(message:<<''reason''>>)".
+
+!4 Ignoring All Tests
+If a fixture throws an exception with a ''class'' name that contains "!style_code(!-IgnoreAllTests-!)", then Slim should ignore and skip all the instructions in the rest of the current batch.   The response for this type of exception should be "!style_code(__EXCEPTION__:IGNORE_ALL_TESTS:)" which may have an optional suffix of: "!style_code(message:<<''reason''>>)".
```
Comment (fhoeben) [inline FitNesseRoot/FitNesse/UserGuide/WritingAcceptanceTests/SliM/SlimProtocol/content.txt:153]: What do you mean with 'current batch', the current test page, or suite, or ...?

File: test/fitnesse/testsystems/slim/SlimTestSystemTableProcessingTest.java [hunk 3]
```diff
@@ -322,7 +340,9 @@ public void kill() {
   private static class IgnoreOnNullPassOtherwiseSlimExpectation implements SlimExpectation {
     @Override
     public TestResult evaluateExpectation(Object returnValues) {
-      return new SlimTestResult(null == returnValues ? ExecutionResult.IGNORE : ExecutionResult.PASS);
+//      if(returnValues.toString().contains("IGNORE_SCRIPT_TEST"))
```
Comment (fhoeben) [inline test/fitnesse/testsystems/slim/SlimTestSystemTableProcessingTest.java:343]: Please remove lines instead of commenting out

**Review by fhoeben — CHANGES_REQUESTED** [review]
A few small remarks and could you please add a description of this change to the ReleaseNotes page?

File: FitNesseRoot/FitNesse/UserGuide/WritingAcceptanceTests/SliM/ExceptionHandling/content.txt (hunk 1 above)
Comment (eireglas) [reply FitNesseRoot/FitNesse/UserGuide/WritingAcceptanceTests/SliM/ExceptionHandling/content.txt]: @fhoeben I used that term as it was used previously for the StopTest exception handling functionality. But I've reworded this line now in the last commit.

File: FitNesseRoot/FitNesse/UserGuide/WritingAcceptanceTests/SliM/SlimProtocol/content.txt (hunk 2 above)
Comment (eireglas) [reply FitNesseRoot/FitNesse/UserGuide/WritingAcceptanceTests/SliM/SlimProtocol/content.txt]: @fhoeben  I used that term as it was used previously for the StopTest exception handling functionality. But I've reworded this line now in the last commit.

File: test/fitnesse/testsystems/slim/SlimTestSystemTableProcessingTest.java (hunk 3 above)
Comment (eireglas) [reply test/fitnesse/testsystems/slim/SlimTestSystemTableProcessingTest.java]: Thanks

**Comment by eireglas** [comment]
@fhoeben I've updated the release notes and the userguide with the last commit. Thanks for reviewing. Let me know if you've any other comments

**Review by fhoeben — APPROVED** [review]

**Comment by eireglas** [comment]
@fhoeben Thanks for merging. Do you have any idea when the next release will be that will include this feature?

**Comment by eireglas** [comment]
@jediwhale Do you know when the next release will be that will include this feature? Thanks.

**Comment by fhoeben** [comment]
I just released `20221219` that should have this included. Unfortunately something went wrong in the release process preventing the version from showing up on the FitNesse site. It is available in Maven Central however.

**Comment by eireglas** [comment]
@fhoeben Great, thanks for letting me know

**Pull request merged**